[NO QA] Add PR review checklists for contributors & C+ (#7565)
Expensify/App: .github/PULL_REQUEST_TEMPLATE.md

- [ ] Verify that no errors appear in the JS console

> ### C+ PR Review Checklist ([full steps here](../PR_REVIEW_CHECKLIST.md))

Reviewer: ```suggestion
### Contributor+ PR Review Checklist ([full steps here](../PR_REVIEW_CHECKLIST.md))
```

> ### Contributor Plus (C+) PR Review Checklist ([full steps here](../PR_REVIEW_CHECKLIST.md))

Author: I like it - I think "guidelines" is a bit less strict sounding but let's see how it feels :) (since as you said... this isn't really a checklist)

<!--

> This is a checklist for Contributor Plus (C+) reviewers. If the PR is internal, these do not need to be checked off, but are still a nice reminder for what to look for when reviewing PRs.

Reviewer: ```suggestion
This is a checklist for Contributor Plus (C+) reviewers. If you are not an Expensify employee or a Contributor+, please leave this section blank, it will be used by the PR reviewer. If you are an Expensify Employee or Contributor+, these do not need to be checked off, but are still a nice reminder for what to look for when reviewing PRs.
```

Author: Hmm I'm pretty sure this checklist is supposed to _only_ be mandatory for C+ and "optional" for internal reviewers, am I wrong?

> This is a checklist for Contributor Plus (C+) reviewers. If the PR is internal, these do not need to be checked off, but are still a nice reminder for what to look for when reviewing PRs.

Author: The current idea in this PR is that the `PR_REVIEW_CHECKLIST` is the real list of things every reviewer needs to check, every time. It's not "guidelines" per say (maybe the "Considerations around code reuse" would be "guidelines").

I tried to make that clear in Slack [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1643980512416919?thread_ts=1643767911.674959&cid=C01SKUP7QR0), and suggested a few alternate solutions but I'm always up for other ideas

-->
- [ ] Are screenshots or videos included for tests on all platforms? ([required](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms))

> - [ ] I (the C+ assigned to review this PR) also tested this PR on **all platforms**

Author: Good call, I think they read better as questions and not first person 👍

- [ ] Is the correct issue linked in the `### Fixed Issues` section?
- [ ] Are testing steps clear and do they cover the changes + possible regressions?
- [ ] Do testing steps cover success & fail scenarios (if applicable)?
- [ ] There are no console errors related to changes in this PR

> - [ ] Appropriate new regression steps are mentioned below (if applicable)

Reviewer: ```suggestion
- [ ] Appropriate new regression steps are mentioned below (if applicable, make sure you tag Expensify/Applauseleads to add these to the regression suite)
```

> - [ ] Code patterns were checked (see `Reviewing the code`)

Reviewer: For this line and the next, can we link directly to the docs section instead of just using code blocks?

Author: Yeah I tried and failed last week, but turns out it was very easy :D

| &nbsp; &nbsp; - I made sure [style guidelines](../STYLING.md) were followed | ❌ / ✅ | ❌ / ✅ |
| I verified that this PR follows the guidelines as stated in the [Review Guidelines](../PR_REVIEW_GUIDELINES.md) | ❌ / ✅ | ❌ / ✅ |

### PR Review Checklist (OPTION 2 - 2 lists, 1 for author 1 for C+)
#### Contributor (PR Author) Checklist
- [ ] I made sure to pull `main` before submitting my PR for review
- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I clearly indicated the environment tests should be run in (Staging vs Production)
- [ ] I wrote testing steps that cover success & fail scenarios (if applicable)
- [ ] I ran the tests & they passed on **all platforms**
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I followed proper code patterns (see [Reviewing the code](../PR_REVIEW_GUIDELINES.md#reviewing-the-code))

>     - [ ] I added comments when appropriate

Reviewer 2: ```suggestion
    - [ ] I added comments when the code was not self explanatory
```

- [ ] Verify that no errors appear in the JS console

> ### Regression Tests to add

Reviewer 2: We could add some guidance here. Something like a commented "Look for flows that may be affected by this change and add tests for those." 

Thoughts?

Reviewer: Good idea! Here's my draft: "If this is a new feature, substantively changes an existing feature, or addresses an edge case, please note the testing steps that should be added to our regular QA checklist."

Author: Hmm I think it's better to leave this off for now since we don't have that process yet (for C+ to view existing regression tests or add to them). I do think it would be nice to be able to offload "adding new regression tests" to contributors / C+ eventually but maybe not at this moment

Author: Done
